cmd/argocd/commands/admin/cluster.go

```diff
@@ -580,9 +585,13 @@ func NewGenClusterConfigCommand(pathOpts *clientcmd.PathOptions) *cobra.Command
 			errors.CheckError(err)
 
 			clst := cmdutil.NewCluster(contextName, clusterOpts.Namespaces, clusterOpts.ClusterResources, conf, bearerToken, awsAuthConf, execProviderConf, labelsMap, annotationsMap)
-			if clusterOpts.InCluster {
+			if clusterOpts.InCluster || clusterOpts.ClusterEndpoint == string(cmdutil.KubeInternalEndpoint) {
```

Comment:
Since we're using this expression at least twice, I was wondering if it makes sense to wrap it up into a method in package cmdutil, e.g. something along the following

```go
func (o ClusterOptions) IsInternalEndpoint() bool {
    return o.InCluster || o.ClusterEndpoint == string(KubeInternalEndpoint)
}
```